### ISSUE TITLE
Install fails

### ISSUE THREAD

**Issue by Kala22** [Lanchon/REPIT#20]
I'm trying to apply REPIT from Cyanogen 12.1 Recovery on my GT-I9100.
I get:

ln: /sbin/linker: Permission denied
/tmp/update_binary: eval: line 1: /tmp/flashize/env/sbin/sh: not found

What could be wrong?

**Comment by Lanchon** [comment]
something's wrong with flashize-env environment replacement. maybe i could change the strategy to a chroot.

i don't care about it though: CM's recovery keeps a lock on the emmc, so even if i fix the environment, repit still won't work.

you need to use TWRP 2 or TWRP 3.

**Comment by Lanchon** [comment]
PS. i cannot do anything about the recovery locking the emmc: only solution is to use a different recovery.

**Comment by Lanchon** [comment]
now documented: e6258c5fe9549ffa6ffb136524c45fb59e311577

**Comment by markusfisch** [comment]
This still happens with the latest TWRP 3 from https://dl.twrp.me/i9100/ on a i9100. There's no /sbin/sh.

Here's recovery.log:

```
Starting recovery on Wed Jan  4 17:42:42 2017

[initialize ui and event]
pixel format -> GGL_PIXEL_FORMAT_BGRA_8888
framebuffer: fd 4 (480 x 800)

[collecting table information]
recovery filesystem table
=========================
  0 '/tmp' 'ramdisk' '(null)' '(null)' 0
  1 '/efs' 'ext4' '/dev/block/mmcblk0p1' '(null)' 0
  2 '/boot' 'emmc' '/dev/block/mmcblk0p2' '(null)' 0
  3 '/recovery' 'emmc' '/dev/block/mmcblk0p6' '(null)' 0
  4 '/cache' 'ext4' '/dev/block/mmcblk0p7' '(null)' 0
  5 '/system' 'ext4' '/dev/block/mmcblk0p9' '(null)' 0
  6 '/data' 'ext4' '/dev/block/mmcblk0p10' '(null)' -16384
  7 '/preload' 'ext4' '/dev/block/mmcblk0p12' '(null)' 0
  8 '/sdcard' 'vfat' '/dev/block/mmcblk1p1' '/dev/block/mmcblk1' 0
  9 '/sdcard1' 'vfat' '/dev/block/mmcblk0p11' '(null)' 0

emmc_checksum fail.

[initialize init-recovery]

#mount /system 
stat() of /dev/block/mmcblk0p9 succeeded on try 1
/system sucessfully mounted

#unmount /cache 
/cache sucessfully unmounted

#exec -f /system/bin/e2fsck -v -y /dev/block/mmcblk0p7 
e2fsck 1.42.6 (21-Sep-2012)
/dev/block/mmcblk0p7: clean, 21/6400 files, 3396/25600 blocks

#insmod -f -v /system/lib/modules/exfat_core.ko 
trace: insmod_main: /system/lib/modules/exfat_core.ko
trace: insmod_main: 1
trace: __out_return: _do_insmod :: can't open '/system/lib/modules/exfat_core.ko' (No such file or directory)

#insmod -f -v /system/lib/modules/exfat_fs.ko 
trace: insmod_main: /system/lib/modules/exfat_fs.ko
trace: insmod_main: 1
trace: __out_return: _do_insmod :: can't open '/system/lib/modules/exfat_fs.ko' (No such file or directory)

#lsmod 
exfat_fs 18096 0 - Live 0xbf014000 (P)
exfat_core 61766 1 exfat_fs, Live 0xbf000000 (P)

#mount /cache 
stat() of /dev/block/mmcblk0p7 succeeded on try 1
/cache sucessfully mounted

#ls /cache/recovery/ 
-rw-r--r-- root     root         7404 2017-01-04 17:20 last_log
-rw-r--r-- root     root            0 2017-01-04 17:20 last_install

#mount /data 
stat() of /dev/block/mmcblk0p10 succeeded on try 1
/data sucessfully mounted

#fcut --limited-file-size=64k -f /data/log/recovery_log.txt /tmp/recovery_backup.txt 
/system sucessfully unmounted
/cache sucessfully unmounted
/data sucessfully unmounted
running init-recovery time : 0.398s

[collecting command]
stat() of /dev/block/mmcblk0p7 succeeded on try 1
I:can't open file '/cache/recovery/command' : No such file or directory 
Command: "/sbin/recovery"
previous_runs = 0
send_intent = (null)
update_package = (null)
att_fota_update = 0
tmo_fota_update = 0
wipe_data = 0, wipe_cache = 0,wipe_sdcard = 0,delete_data = 0,wipe_data_crypto = 0,wipe_carrier = 0
show_ui_text = 0
update_factory_csc = 0, update_home_csc = 0, factory_fota = 0
carry_out = (null)


[property list]
ro.boot.serialno=00097c4358e36f
ro.sec_debug.enable=0
ro.serialno=00097c4358e36f
ro.bootmode=unknown
ro.baseband=unknown
ro.bootloader=unknown
ro.debug_level=unknown
ro.emmc_checksum=unknown
ril.dualmode.change=unknown
ro.hardware=smdk4210
ro.revision=14
ro.factorytest=0
ro.uart_debug=0
ro.secure=1
ro.allow.mock.location=0
ro.debuggable=0
persist.sys.usb.config=mtp
init.svc.ueventd=running
init.svc.recovery=running

stat() of /dev/block/mmcblk0p1 succeeded on try 1
stat() of /dev/block/mmcblk0p9 succeeded on try 1
stat() of /dev/block/mmcblk0p1 succeeded on try 1
read csc code = 'DBT'

# MANUAL MODE #
No command specified.

[installing multi-csc]

#echo 

#echo -- Appling Multi-CSC... 
-- Appling Multi-CSC...

#mount /system 
stat() of /dev/block/mmcblk0p9 succeeded on try 1
/system sucessfully mounted

#echo Applied the CSC-code : DBT 
Applied the CSC-code : DBT

#cp -y -f -r -v /system/csc/common / 
trace: __out_return: __scan_cp_dir :: cannot get src '/system/csc/common'  (No such file or directory)

#unmount /system 
/system sucessfully unmounted

#mount /system 
stat() of /dev/block/mmcblk0p9 succeeded on try 1
/system sucessfully mounted

#cmp -r /system/csc/common / 
trace: __out_return: __scan_cmp_dir :: Can't open dir '/system/csc/common' .. skip (No such file or directory)

#cp -y -f -r -v /system/csc/DBT/system /system 
trace: __out_return: __scan_cp_dir :: cannot get src '/system/csc/DBT/system'  (No such file or directory)

#rm -v /system/csc_contents 
removed symbolic '/system/csc_contents' 

#ln -v -s /system/csc/DBT/csc_contents /system/csc_contents 
create symbolic '/system/csc/DBT/csc_contents' -> '/system/csc_contents'

#unmount /system 
/system sucessfully unmounted

#mount /system 
stat() of /dev/block/mmcblk0p9 succeeded on try 1
/system sucessfully mounted

#cmp -r /system/csc/DBT/system /system 
trace: __out_return: __scan_cmp_dir :: Can't open dir '/system/csc/DBT/system' .. skip (No such file or directory)

#rm -v --limited-file-size=0 /system/app/* 

#echo Successfully applied multi-CSC. 
Successfully applied multi-CSC.
/system sucessfully unmounted
running multi-csc time : 0.556s

[installing post-recovery]

#mount /data 
stat() of /dev/block/mmcblk0p10 succeeded on try 1
/data sucessfully mounted

#mkdir system log 0775 /data/log 

#cp -y -f -v /tmp/recovery_backup.txt /data/log/recovery_log.txt 
/tmp/recovery_backup.txt -> /data/log/recovery_log.txt (5558 bytes) 
transferred in writed time : 0.000s closed time : 0.035s progressed with 4096.00M byte/sec
src : -rw-rw-r-- root     root         5558 2017-01-04 17:42 recovery_backup.txt
dst : -rw-rw-r-- root     root         5558 2017-01-04 17:42 recovery_log.txt

/data sucessfully unmounted
running post-recovery time : 0.104s
stat() of /dev/block/mmcblk0p10 succeeded on try 1
create recovery log file '/data/log/recovery_last_kernel_log.txt' 
create kernel log file '/data/log/recovery_kernel_log.txt' 
create recovery log file '/data/log/recovery_log.txt' 
stat() of /dev/block/mmcblk0p9 succeeded on try 1
void copy_log_file(char const*, char const*, int) :: create recovery log file '/cache/recovery/log' 
void copy_log_file(char const*, char const*, int) :: create recovery log file '/cache/recovery/last_log' 
void copy_log_file(char const*, char const*, int) :: create recovery log file '/cache/recovery/last_install' 
key press -> key (114), visible(1)
key press -> key (114), visible(1)
key press -> key (116), visible(1)
stat() of /dev/block/mmcblk0p7 succeeded on try 1
stat() of /dev/block/mmcblk1p1 succeeded on try 1
key press -> key (114), visible(1)
key press -> key (114), visible(1)
key press -> key (116), visible(1)

-- Install /sdcard ...
Finding update package...
I:Update location: /tmp/sideload/package.zip
Opening update package...
I:2 key(s) loaded from /res/keys
Verifying update package...
I:comment is 1738 bytes; signature 1720 bytes from end
signed len : 823623 
sha1 : 823680
b9a07298 d9e7ab35 6c64e4a4 5d2413e9 12ea417a 
b9 a0 72 98 d9 e7 ab 35 6c 64 e4 a4 5d 24 13 e9 12 ea 41 7a 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 64 8a 38 
I:whole-file signature verified against key 0
I:verify_file returned 0
Installing update...
E:Can't run /tmp/update_binary (No such file or directory)
E :Error in /tmp/sideload/package.zip
(Status 255)
E:Error in /tmp/sideload/package.zip
(Status 255)
key press -> key (116), visible(1)

-- Install /sdcard ...
stat() of /dev/block/mmcblk1p1 succeeded on try 1
stat() of /dev/block/mmcblk0p7 succeeded on try 1
Finding update package...
I:Update location: /tmp/sideload/package.zip
Opening update package...
I:2 key(s) loaded from /res/keys
Verifying update package...
I:comment is 1738 bytes; signature 1720 bytes from end
signed len : 823623 
sha1 : 823680
b9a07298 d9e7ab35 6c64e4a4 5d2413e9 12ea417a 
b9 a0 72 98 d9 e7 ab 35 6c 64 e4 a4 5d 24 13 e9 12 ea 41 7a 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 64 8a 38 
I:whole-file signature verified against key 0
I:verify_file returned 0
Installing update...
E:Can't run /tmp/update_binary (No such file or directory)
E :Error in /tmp/sideload/package.zip
(Status 255)
E:Error in /tmp/sideload/package.zip
(Status 255)
key press -> key (116), visible(1)

-- Install /sdcard ...
stat() of /dev/block/mmcblk1p1 succeeded on try 1
stat() of /dev/block/mmcblk0p7 succeeded on try 1
Finding update package...
I:Update location: /tmp/sideload/package.zip
Opening update package...
I:2 key(s) loaded from /res/keys
Verifying update package...
I:comment is 1738 bytes; signature 1720 bytes from end
signed len : 823623 
sha1 : 823680
b9a07298 d9e7ab35 6c64e4a4 5d2413e9 12ea417a 
b9 a0 72 98 d9 e7 ab 35 6c 64 e4 a4 5d 24 13 e9 12 ea 41 7a 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 64 8a 38 
I:whole-file signature verified against key 0
I:verify_file returned 0
Installing update...
E:Can't run /tmp/update_binary (No such file or directory)
E :Error in /tmp/sideload/package.zip
(Status 255)
E:Error in /tmp/sideload/package.zip
(Status 255)
key press -> key (116), visible(1)

-- Install /sdcard ...
stat() of /dev/block/mmcblk1p1 succeeded on try 1
stat() of /dev/block/mmcblk0p7 succeeded on try 1
Finding update package...
I:Update location: /tmp/sideload/package.zip
Opening update package...
I:2 key(s) loaded from /res/keys
Verifying update package...
I:comment is 1738 bytes; signature 1720 bytes from end
signed len : 823623 
sha1 : 823680
b9a07298 d9e7ab35 6c64e4a4 5d2413e9 12ea417a 
b9 a0 72 98 d9 e7 ab 35 6c 64 e4 a4 5d 24 13 e9 12 ea 41 7a 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 64 8a 38 
I:whole-file signature verified against key 0
I:verify_file returned 0
Installing update...
E:Can't run /tmp/update_binary (No such file or directory)
E :Error in /tmp/sideload/package.zip
(Status 255)
E:Error in /tmp/sideload/package.zip
(Status 255)
```

**Comment by Lanchon** [comment]
piggybacking on old issue is NOT cool !!!

- no kernel info
- no REPIT version info
- no flashing procedure

so i can't help you.
read the docs, then open an issue according to docs if necessary.
thanks!
